### ISSUE TITLE
impute() does not handle non missing values in columns

### ISSUE THREAD

**Issue by jakob-r** [mlr-org/mlr#848]
impute should be able to do nothing if no missing values are present:
```r
library(mlr)
iris.m = iris
imp = impute(data = iris.m, 
  target = "Species", 
  cols = list(Sepal.Width = imputeLearner(makeLearner("regr.rpart"))))
#Error in impute(data = iris.m, target = "Species", cols = list(Sepal.Width = imputeLearner(makeLearner("regr.rpart")))) : 
#  Assertion on 'newdata' failed: Must have at least 1 rows, but has 0 rows.

**Comment by florianfendt** [comment]
Well, ````impute()```` in general is not the problem here, since with e.g. ````imputeMode()```` and ````imputeMedian()```` the above code works fine.
So imho the problem is only the ```ìmputeLearner()```` method since we then want to predict on an empty data set.

**Comment by berndbischl** [comment]
the issue #893 contains another piece of code that we can check in a unit test

**Comment by berndbischl** [comment]
fixed in PR #1037
